internals/maintenance_scripts.py

```diff
@@ -529,6 +529,8 @@ def handle_creation(self, stage: Stage, stage_dict: StageDict) -> str | None:
                       f'{stage.key.integer_id()}')
       cloud_tasks_helpers.enqueue_task(
           '/tasks/email-ot-creation-request-failed', {'stage': stage_dict})
+      stage.ot_setup_status = OT_CREATION_FAILED
+      stage.put()
```

Comment:
I think the retry logic above can be improved:
- Based on the discussions in the design doc, the request should be retried only if there is a transient error from the OT server. The catch logic should be more granular, based on the exception types.
- attempted_requests += 1 under the `except error` seems unnecessary. It basically means that we are only gonna retry twice. Then we might as well remove the while loop and attempted_requests, as it is not providing any values
- create_origin_trial is throwing an exception when the response code is not 200, which means request_success is false. `request_success` can be removed and nested all the logging and email sending to `except error`.
- According the error handling section in the design doc, more granular error handling is needed here, based on the response code. Lets leave a TODO here

Reply:
I've removed the retry logic and added a TODO here 👍